app/src/main/java/net/vonforst/evmap/fragment/MapFragment.kt

```diff
@@ -566,29 +566,31 @@ class MapFragment : Fragment(), OnMapReadyCallback, MapsActivity.FragmentCallbac
             }
         })
         vm.chargepoints.observe(viewLifecycleOwner, Observer { res ->
-            when (res.status) {
-                Status.ERROR -> {
-                    val view = view ?: return@Observer
-
-                    connectionErrorSnackbar?.dismiss()
-                    connectionErrorSnackbar = Snackbar
-                        .make(view, R.string.connection_error, Snackbar.LENGTH_INDEFINITE)
-                        .setAction(R.string.retry) {
-                            connectionErrorSnackbar?.dismiss()
-                            vm.reloadChargepoints()
-                        }
-                    connectionErrorSnackbar!!.show()
-                }
-                Status.SUCCESS -> {
-                    connectionErrorSnackbar?.dismiss()
-                }
-                Status.LOADING -> {
-                }
-            }
 
             val chargepoints = res.data
             if (chargepoints != null) {
                 updateMap(chargepoints)
+                when (res.status) {
+                    Status.ERROR -> {
```

Comment:
Like this, the connection error snackbar will never be shown when `chargepoints == null && res.status == Status.ERROR` (i.e. if no cached data is available and an error occurred when loading the data)...